CameraTools/CamTools.cs

```diff
@@ -3617,6 +3657,32 @@ private FieldInfo GetAITargetField()
 			return null;
 		}
 
+		private void UpdatebdWMVesselList()
```

Comment:
I think it'd be better to leverage the `LoadedVesselSwitcher` instance to get this information.
I've added a `Vessels` property to the `LoadedVesselSwitcher` instance in BDArmory and the corresponding `GetBDVessels` function in CameraTools that gets a dictionary of the active vessels keyed by teams (in case you want to limit the centroid to a specific team).

This also avoids updating `wmComponent` (which `CheckForBDWM` modifies) which is used to check for incoming threats to the current active vessel.

Reply:
Grabbing the loadedvesselSwitcher list directly instead of coming up with a copy would be much more straightforward, yes.
And linking it to teams would be a good addition. Will see about getting stuff switched over to the new method.